src/components/UserForm.jsx

```diff
@@ -0,0 +1,300 @@
+          : value,
+    }));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (!isEditing && formData.password !== formData.confirmPassword) {
+      alert("Passwords do not match");
+      return;
+    }
+
+    if (!agreedToTerms && !isEditing) {
+      alert("Please agree to the terms and conditions");
+      return;
+    }
+
+    // Pass form data to the parent component for hadeling(create/update)
+    onSubmit(formData);
+  };
+
+  return (
+    <div
+      className="min-h-screen py-20"
+      style={{ backgroundImage: "linear-gradient(115deg, #9F7AEA, #FEE2FE)" }}
+    >
+      <div className="container mx-auto ">
+        <div className="w-full py-16 px-12 bg-white bg-opacity-80 relative z-10">
+          <div
+            className="w-full md:w-8/12 bg-white rounded-xl mx-auto p-6 shadow-lg overflow-hidden"
+            style={{
+              backgroundImage: `url(${noQicon})`,
+              backgroundSize: "contain", // Adjust size (you can use "cover", "contain", or a specific value)
+              backgroundPosition: "center",
+              backgroundRepeat: "no-repeat",
+              opacity: "0.1",
+            }}
+          />
+          <form onSubmit={handleSubmit}>
+            <div className="grid grid-cols-2 gap-5">
+              {/* <div className="w-1/2 py-16 px-12"> */}
+              <input
+                type="text"
+                name="first_name"
+                value={formData.first_name}
+                onChange={handleChange}
+                placeholder="Firts Name"
+                required
+                className="border rounded border-gray-400 py-1 px-2"
+              />
+              <input
+                type="text"
+                name="last_name"
+                value={formData.last_name}
+                onChange={handleChange}
+                placeholder="Last Name"
+                className="border rounded border-gray-400 py-1 px-2 "
+              />
+
+              <div className="mt-5">
+                <input
+                  type="email"
+                  name="email"
+                  value={formData.email}
+                  onChange={handleChange}
+                  placeholder="Email"
+                  className="border rounded border-gray-400 py-1 px-3 w-full"
+                />
+              </div>
+              <div className="mt-5">
+                <input
+                  type="text"
+                  name="phone"
+                  value={formData.phone}
+                  onChange={handleChange}
+                  placeholder="Phone"
+                  className="border rounded border-gray-400 py-1 px-3 w-full"
+                />
+              </div>
+
+              <div className="mt-5 relative">
+                <CustomDropdown
+                  name="gender"
+                  value={formData.gender}
+                  onChange={(value) =>
+                    setFormData((prevData) => ({ ...prevData, gender: value }))
```

Comment:
Jag fick förljande error när jag ändrade kön:
`Uncaught Error: Objects are not valid as a React child (found: object with keys {target}). If you meant to render a collection of children, use an array instead.
    at throwOnInvalidObjectType (chunk-IA5K47JS.js?v=decbacf6:11202:17)
    at reconcileChildFibers2 (chunk-IA5K47JS.js?v=decbacf6:11832:15)
    at reconcileChildren (chunk-IA5K47JS.js?v=decbacf6:14274:37)
    at updateHostComponent (chunk-IA5K47JS.js?v=decbacf6:14782:11)
    at beginWork (chunk-IA5K47JS.js?v=decbacf6:15903:22)
    at beginWork$1 (chunk-IA5K47JS.js?v=decbacf6:19721:22)
    at performUnitOfWork (chunk-IA5K47JS.js?v=decbacf6:19166:20)
    at workLoopSync (chunk-IA5K47JS.js?v=decbacf6:19105:13)
    at renderRootSync (chunk-IA5K47JS.js?v=decbacf6:19084:15)
    at recoverFromConcurrentError (chunk-IA5K47JS.js?v=decbacf6:18704:28)`